sycl/test/on-device/xocc/simple_tests/kernel_argument_forwarding.cpp

```diff
@@ -22,19 +22,18 @@ int main() {
   Queue.submit([&](sycl::handler &cgh) {
     auto Accessor = Buffer.get_access<sycl::access_mode::write>(cgh);
     cgh.single_task<class FirstKernel>(
-        sycl::xilinx::kernel_param("--optimize 2"_cstr, [=] {
+        sycl::xilinx::kernel_param([=] {
           // CHECK1: v++ {{.*}}class_FirstKernel{{.*}} --optimize 2
           Accessor[0] = 0;
-        }));
+        }, "--optimize 2"_cstr));
```

Comment:
The problem with the decoration at the end is that it is less evident there is a decoration, even if it is arguable that the most interesting part is the kernel itself.

Reply:
I personally find it easier to parse : the decorator name is still the first thing that comes after the `single_task<>(` so we know the kernel is decorated, and if we care we can find the detail of the decoration after the kernel.

It is also easier to handle variadic parameters this way, but this is not a blocking issue if the alternative is preferable.

Reply:
We can keep it as is for now.
Parsing would be easier by combining a property concept + variadic pack.